### ISSUE TITLE
Allow Rubocop versions above 1.18.x?

### ISSUE THREAD

**Issue by emmahsax** [jekyll/rubocop-jekyll#23]
The requirements of this gem require [rubocop `~> 1.18.0`](https://github.com/jekyll/rubocop-jekyll/blob/1b0f945844788c164f9eed989ba977200a427dc4/rubocop-jekyll.gemspec#L22), meaning that Rubocop versions above `1.19.0` aren't allowed.

However, Rubocop doesn't support Ruby 3.2 until [Rubocop version `1.26.0`](https://github.com/rubocop/rubocop/blob/v1.26.0/lib/rubocop/target_ruby.rb) Is there any reason we can't allow Rubocop 1.26 or higher?

```bash
$ bundle exec rubocop
Error: RuboCop found unknown Ruby version 3.2 in `.ruby-version`.
Supported versions: 2.5, 2.6, 2.7, 3.0, 3.1
```

**Comment by ashmaroli** [comment]
Hello @emmahsax,
The reason this gem locks to a certain RuboCop version is because RuboCop releases new cops every now and then, often failing CI on lines of code unrelated to tested commit / pull request.
That said, there is no reason blocking this plugin from getting itself updated to allow upto RuboCop 1.26.x, other than lack of need / motivation.

**Comment by parkr** [comment]
I think we should upgrade. There's a rubocop failure in https://github.com/jekyll/github-metadata/pull/245 that is an internal error within Rubocop. Since this is locked to 1.18.x, I can't test if it's fixed in a later version.

**Comment by parkr** [comment]
> The reason this gem locks to a certain RuboCop version is because RuboCop releases new cops every now and then, often failing CI on lines of code unrelated to tested commit / pull request.

Might be worthwhile changing our config to disable new cops if we're worried about this:

```yaml
AllCops:
    NewCops: enable
```

Either way, we can stay more up-to-date if we release this more often. That's made easier now by the addition of the release workflow, so we could potentially bump our rubocop dependency, then release a new version of rubocop-jekyll and keep the dependency fixed on a single minor version. Then, dependabot-based upgrades (at least for the plugin repos) would be able to get the upgrades but handle them in a PR.

**Comment by parkr** [comment]
Fixed by #27.

**Comment by parkr** [comment]
Follow #30 for updates. Everything is ready to go, but I can't push the update. Hold tight!
